code/datums/components/food/edible.dm

```diff
@@ -308,7 +309,31 @@ Behavior that's still missing from this component that original food items had t
 	var/fullness = eater.get_fullness() + 10 //The theoretical fullness of the person eating if they were to eat this
 
 	if(eater == feeder)//If you're eating it yourself.
+		var/area/current_area = get_area(eater)
+		var/ate_in_service = (current_area.times_eaten_counter >= BREAK_ROOM_DESIGNATION)
+		var/ate_at_table = find_adjacent_tables(eater)
+		var/ate_with_chair = eater.buckled
+		var/ate_with_utensils = eater.is_holding_item_of_type(/obj/item/kitchen)
+		if(ate_in_service)
+			eat_time *= 0.5
+		else
+			eat_time *= 1.5
+		if(!(food_flags & FOOD_FINGER_FOOD))
+			if(ate_with_utensils)
+				eat_time *= 0.75
+			else
+				eat_time *= 1.25
+			if(ate_at_table)
+				eat_time *= 0.75
+			else
+				eat_time *= 1.25
+			if(ate_with_chair)
+				eat_time *= 0.75
+			else
+				eat_time *= 1.25
```

Comment:
could this logic be tied to the eater itself? surely the mob type should be able to have some say in how it impacts eating time
that and early returns would make this code much easier to read

Reply:
it's a component, everything has to be self-contained

Reply:
surely the eat delay of the mob should be something we ask the mob about
it's not like the mob needs to know where the request is coming from, right?

Reply:
eat delay is a per food item thing, not on the mob

Reply:
why are we using so much info from the eater then?
~~also why do those vars exist, as comments?~~